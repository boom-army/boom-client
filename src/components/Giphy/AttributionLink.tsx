import React, { useContext } from "react";
import Link from "@mui/material/Link";
import { styled } from "@mui/system";
import { ThemeContext } from "../../contexts/theme";
import { darkTheme } from "../../styles/themes";

export const AttributionLink: React.FC<{ src: any |null}> = ({ src }) => {
  const { theme } = useContext(ThemeContext);
  const isDarkTheme = theme.primaryColor === darkTheme.primaryColor;

  const GiphyAttributionImg = styled("img")({
    boxSizing: "content-box",
    borderBottom: "2px solid transparent",
    paddingBottom: "3px",
    "&:hover": {
      borderBottom: `2px solid ${theme.tertiaryColor}`,
    },
  });

  return (
    <Link href={src} target="_blank">
      <GiphyAttributionImg
        width="150px"
        src={`/assets/giphy-${isDarkTheme ? "dark" : "light"}.png`}
        alt="Powered by Giphy"
      />
    </Link>
  );
};
