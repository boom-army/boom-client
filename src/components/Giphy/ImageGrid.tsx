import React from "react";
import ImageList from "@mui/material/ImageList";
import ImageListItem from "@mui/material/ImageListItem";
import { Box } from "@mui/system";
import { GIFObject } from "giphy-api";
import { Loader } from "../Loader";
import Stack from "@mui/material/Stack";

interface Props {
  gifArr: GIFObject[];
  setGif: React.Dispatch<React.SetStateAction<GIFObject>>;
  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
  isLoadingMore: boolean;
}

export const ImageGrid: React.FC<Props> = ({
  gifArr,
  setGif,
  setOpen,
  isLoadingMore,
}) => {
  if (!gifArr?.length) {
    return (
      <Box sx={{ margin: 2 }}>
        <p>No gifs were found</p>
      </Box>
    );
  }
  return (
    <Stack direction="column">
      <ImageList variant="masonry" cols={3} gap={8} sx={{ overflowY: "auto" }}>
        {gifArr.map((item) => (
          <ImageListItem
            component="button"
            key={item.id}
            onClick={() => {
              setGif(item);
              setOpen(false);
            }}
            sx={{
              margin: 0,
              padding: 0,
              appearance: "none",
              background: "none",
              border: "none",
              lineHeight: 0,
            }}
          >
            <img
              src={item.images.fixed_width_downsampled.url}
              alt={item.title}
            />
          </ImageListItem>
        ))}
      </ImageList>
      <Box height="50px">{isLoadingMore && <Loader />}</Box>
    </Stack>
  );
};
