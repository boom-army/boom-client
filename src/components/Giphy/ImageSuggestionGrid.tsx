import React from "react";
import ImageList from "@mui/material/ImageList";
import ImageListItem from "@mui/material/ImageListItem";
import ImageListItemBar from "@mui/material/ImageListItemBar";
import { GIPHY_SEARCH_SUGGESTIONS } from "../../constants/giphy-search-suggestions";

type SetInput = React.Dispatch<React.SetStateAction<string>>;
type SetIsLoading = React.Dispatch<React.SetStateAction<boolean>>;

export const ImageSuggestionGrid: React.FC<{
  setInput: SetInput;
  setIsLoading: SetIsLoading;
}> = ({ setInput, setIsLoading }) => {
  return (
    <>
      <ImageList cols={2} gap={2} sx={{ overflowY: "hidden" }}>
        {GIPHY_SEARCH_SUGGESTIONS.map((item, index) => {
          // If the last item is odd, then make it fill the entire row
          const isLastItem = index === GIPHY_SEARCH_SUGGESTIONS.length - 1;
          const isLastItemOdd = GIPHY_SEARCH_SUGGESTIONS.length % 2 === 1;
          const cols = isLastItem && isLastItemOdd ? 2 : 1;
          const rows = isLastItem && isLastItemOdd ? 2 : 1;

          return (
            <ImageListItem
              key={item.id}
              cols={cols}
              rows={rows}
              sx={{
                height: "150px!important",
                overflow: "hidden",
                cursor: "pointer",
              }}
              onClick={() => {
                setIsLoading(true);
                setInput(item.title);
              }}
            >
              <img src={item.image} alt={item.title} loading="lazy" />
              <ImageListItemBar
                sx={{
                  background:
                    "linear-gradient(to bottom, rgba(0,0,0,0.7) 0%, " +
                    "rgba(0,0,0,0.3) 70%, rgba(0,0,0,0) 100%)",
                }}
                title={item.title}
                position="top"
                actionPosition="left"
              />
            </ImageListItem>
          );
        })}
      </ImageList>
    </>
  );
};
