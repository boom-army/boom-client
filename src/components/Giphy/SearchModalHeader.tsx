import React, { useContext } from "react";
import CloseIcon from "@mui/icons-material/Close";
import IconButton from "@mui/material/IconButton";
import { FormControl, InputAdornment, useTheme } from "@mui/material";
import { TextField } from "@mui/material";

interface Props {
  input: string;
  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
  setInput: React.Dispatch<React.SetStateAction<string>>;
  handleClose: () => void;
}

export const SearchModalHeader: React.FC<Props> = ({
  input,
  setIsLoading,
  setInput,
  handleClose,
}) => {
  const theme = useTheme();
  return (
    <FormControl fullWidth={true}>
      <TextField
        hiddenLabel={true}
        fullWidth={true}
        type="text"
        autoFocus
        placeholder="Search for GIF"
        value={input}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
          setIsLoading(true);
          setInput(e.target.value);
        }}
        InputProps={{
          style: { color: theme.palette.primary, padding: 10 },
          endAdornment: (
            <InputAdornment position="end">
              <IconButton
                onClick={handleClose}
                aria-label="close"
                size="medium"
                disableRipple={true}
              >
                <CloseIcon sx={{ color: theme.accentColor }} />
              </IconButton>
            </InputAdornment>
          ),
        }}
      />
    </FormControl>
  );
};
