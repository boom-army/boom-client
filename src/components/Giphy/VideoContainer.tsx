import React, { useRef } from "react";
import { styled } from "@mui/material/styles";
import { Box } from "@mui/system";
import IconButton from "@mui/material/IconButton";
import CloseIcon from "@mui/icons-material/Close";

interface Gif {
  id?: string;
  fixedHeightUrl?: string;
  originalUrl?: string;
  title?: string;
  url?: string;
}

interface VideoProps {
  gif: Gif;
  onClose?: (() => void) | undefined;
}

const VideoBoxWrapper = styled("div")(({ theme }) => ({
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  justifyContent: "center",
  position: "relative",
  borderRadius: "16px",
  border: `1px solid ${theme.palette.secondary}`,
  width: "90%",
  margin: "0 0 0.75rem",
  overflow: "hidden",
}));

const Video = styled("video")({
  width: "100%",
  cursor: "auto",
});

export const VideoContainer = React.memo(({ gif, onClose }: VideoProps) => {
  const { fixedHeightUrl, title } = gif;
  const videoRef = useRef<HTMLVideoElement>(null);

  const handleVideoPlayerClick = () => {
    const video = videoRef.current;
    if (video && video.paused) {
      video.play().catch((error) => {
        console.error("Video playback failed:", error);
      });
    }
  };

  return (
    <VideoBoxWrapper>
      {onClose && (
        <Box
          sx={{
            position: "absolute",
            left: "3%",
            top: "3%",
            padding: "5px",
            borderRadius: "50%",
            background: "rgba(0, 0, 0, 0.7)",
            lineHeight: "1.5",
            fontSize: "14px",
            zIndex: "99",
            "&:hover": {
              background: "rgba(0, 0, 0, 0.57)",
            },
          }}
        >
          <IconButton onClick={onClose} aria-label="close" size="medium">
            <CloseIcon />
          </IconButton>
        </Box>
      )}

      <Video
        ref={videoRef}
        onClick={handleVideoPlayerClick}
        playsInline
        autoPlay
        muted
        loop
        aria-label={title}
      >
        <source src={fixedHeightUrl} type="video/mp4" />
      </Video>

      <Box
        sx={{
          position: "absolute",
          left: "3%",
          bottom: "3%",
          padding: "0 4px",
          borderRadius: "4px",
          background: "rgba(0, 0, 0, 0.77)",
          lineHeight: "1.5",
          fontSize: "14px",
        }}
      >
        GIF
      </Box>
    </VideoBoxWrapper>
  );
});
