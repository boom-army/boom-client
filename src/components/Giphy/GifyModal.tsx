import React, {
  useState,
  useEffect,
  useContext,
  useCallback,
  useRef,
} from "react";
import { Box } from "@mui/system";
import { GifIcon } from "../Icons";
import { GiphyContext, Search } from "../../contexts/giphy";
import { ImageGrid } from "./ImageGrid";
import { ImageSuggestionGrid } from "./ImageSuggestionGrid";
import { Loader } from "../Loader";
import { SearchModalHeader } from "./SearchModalHeader";
import { useTheme } from '@mui/material/styles';
import { debounce } from "lodash";
import { IconButton, Stack, Modal } from "@mui/material";
import { blue, grey } from "@mui/material/colors";

const GIPHY_API = process.env.REACT_APP_GIPHY_KEY as string;

const queryGiphy = async (
  query: string,
  offset: number = 0
): Promise<Search> => {
  const queryParams = new URLSearchParams({
    api_key: GIPHY_API,
    offset: offset.toString(),
    q: query,
  });
  const response = await fetch(
    `https://api.giphy.com/v1/gifs/search?${queryParams}`
  );
  const json = await response.json();
  return { query, gif: json.data };
};

export const GifyModal: React.FC<{
  setGif: any;
}> = ({ setGif }) => {
  const [input, setInput] = useState("");
  const [open, setOpen] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [isLoadingMore, setIsLoadingMore] = useState(false);
  const [error, setError] = useState<any>(null);
  const [searchResults, setSearchResults] = useState<Search | null>(null); // current searched gifs
  const { searchGiphyCache, setSearchGiphyCache } = useContext(GiphyContext); // search cache
  const theme = useTheme();
  const handleClose = () => setOpen(false);
  const handleOpen = () => {
    setInput("");
    setError(false);
    setSearchResults(null);
    setOpen(true);
  };

  const imageBoxRef = useRef<any>(null);

  const handleScroll = debounce(async () => {
    if (!imageBoxRef.current || !input) return;
    const { scrollHeight, scrollTop, clientHeight } = imageBoxRef.current;

    if (scrollHeight - scrollTop - clientHeight >= 1) return;

    try {
      const currentSearchResultsLength = searchResults?.gif.length;
      // the initial search returned no hits so there is no point querying again
      if (currentSearchResultsLength === 0) return;

      setIsLoadingMore(true);

      const query = input.toLowerCase();
      const result = await queryGiphy(query, currentSearchResultsLength);

      if (!result?.gif.length) return;

      const updatedSearchCache = searchGiphyCache.map((cache) => {
        if (cache.query === result.query) {
          return {
            query: cache.query,
            gif: [...cache.gif, ...result.gif],
          };
        }
        return cache;
      });

      setSearchGiphyCache(updatedSearchCache);
      const updatedGifs =
        updatedSearchCache.find((cache) => cache.query === query) || null;
      setSearchResults(updatedGifs);
    } finally {
      setIsLoadingMore(false);
    }
  }, 300);

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const memoDebouncedScrollHandler = useCallback(handleScroll, [
    searchResults,
    searchGiphyCache,
    input,
  ]);

  useEffect(() => {
    const imageListNode = imageBoxRef.current;

    if (!imageListNode) return;

    imageListNode.addEventListener("scroll", memoDebouncedScrollHandler);
    return () => {
      imageListNode.removeEventListener("scroll", memoDebouncedScrollHandler);
    };
  }, [memoDebouncedScrollHandler]);

  const handleInputQuery = debounce(
    async (value: string, searchGiphyCache: Search[]) => {
      setError(false);

      const query = value.toLowerCase();

      if (!query) {
        setIsLoading(false);
        return setSearchResults(null);
      }

      const cachedGif = searchGiphyCache.find((gif) => gif.query === query);

      if (cachedGif) {
        setIsLoading(false);
        return setSearchResults(cachedGif);
      }

      try {
        const searchResult = await queryGiphy(query);
        setSearchGiphyCache([...searchGiphyCache, searchResult]);
        setSearchResults(searchResult);
        setError(false);
      } catch (error) {
        setError(true);
      } finally {
        setIsLoading(false);
      }
    },
    800
  );

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const memoDebouncedQuery = useCallback(handleInputQuery, []);

  useEffect(() => {
    memoDebouncedQuery(input, searchGiphyCache);
  }, [input, memoDebouncedQuery, searchGiphyCache]);

  return (
    <>
      <IconButton onClick={handleOpen}>
        <GifIcon />
      </IconButton>
      <Modal
        open={open}
        onClose={handleClose}
      >
        <Box
          sx={{
            position: "absolute",
            top: "30%",
            transform: "translate(20%, -60%)",
            bgcolor: theme.palette.background.default,
            border: `2px solid ${blue[100]}`,
            borderRadius: 1,
            maxWidth: "90vw",
            width: 500,
            height: 650,
            minHeight: 500,
            maxHeight: "90vh",
          }}
        >
          <Stack direction="column" sx={{ height: "100%" }}>
            <SearchModalHeader
              input={input}
              setIsLoading={setIsLoading}
              setInput={setInput}
              handleClose={handleClose}
            />
            <Box
              ref={imageBoxRef}
              sx={{ height: "100%", overflowY: "auto", overflowX: "hidden" }}
            >
              {isLoading || error ? (
                <Box sx={{ margin: 2, position: "relative" }}>
                  {isLoading ? (
                    <Loader />
                  ) : (
                    <p>There was an error while searching.</p>
                  )}
                </Box>
              ) : (
                <>
                  {searchResults && (
                    <ImageGrid
                      gifArr={searchResults.gif}
                      setGif={setGif}
                      setOpen={setOpen}
                      isLoadingMore={isLoadingMore}
                    />
                  )}
                  {!searchResults && (
                    <ImageSuggestionGrid
                      setInput={setInput}
                      setIsLoading={setIsLoading}
                    />
                  )}
                </>
              )}
            </Box>
          </Stack>
        </Box>
      </Modal>
    </>
  );
};
