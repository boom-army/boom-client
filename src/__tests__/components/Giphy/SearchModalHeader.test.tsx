import React from "react";
import { SearchModalHeader } from "../../../components/Giphy/SearchModalHeader";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { createSerializer } from '@emotion/jest'

expect.addSnapshotSerializer(createSerializer(
  {
    classNameReplacer(className, index) {
      return `css-${index}`
    },
        // includeStyles: false 
  }
));

let setIsLoading: any = false;
let setInput: any = "";
let handleClose: any;

describe("<SearchModalHeader/> component :", () => {
  test("display <SearchModalHeader/> component  when Search result found ", async () => {
    let rendered = render(
      <SearchModalHeader
        input={"anger"}
        setIsLoading={setIsLoading}
        setInput={setInput}
        handleClose={handleClose}
      />
    );
    expect(rendered).toMatchSnapshot();
  });

  test("display <SearchModalHeader/> component  when Search result found ", async () => {
    let rendered = render(
      <SearchModalHeader
        input={"anger"}
        setIsLoading={setIsLoading}
        setInput={setInput}
        handleClose={handleClose}
      />
    );
    fireEvent.click(rendered.getByPlaceholderText("Search"));
    expect(rendered.asFragment()).toMatchSnapshot();
  });
});

test("Assertion testing of<SearchModalHeader/> component ", () => {
  render(
    <SearchModalHeader
      input={"anger"}
      setIsLoading={setIsLoading}
      setInput={setInput}
      handleClose={handleClose}
    />
  );
  expect(screen.getByRole("button", { name: "close" })).toBeInTheDocument();
  expect(screen.getByText("Search for gif")).toBeInTheDocument();
});
