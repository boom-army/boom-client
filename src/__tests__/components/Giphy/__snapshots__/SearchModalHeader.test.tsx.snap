// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchModalHeader/> component : display <SearchModalHeader/> component  when Search result found  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .css-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 10px 16px;
}

.css-0>:not(style)+:not(style) {
  margin: 0;
  margin-left: 16px;
}

.css-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.css-1::-moz-focus-inner {
  border-style: none;
}

.css-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .css-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.css-1.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.css-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.css-3 {
  width: 315px;
  background: undefined;
  padding: 0.2rem 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom: 1px solid undefined;
  margin-bottom: 2rem;
}

.css-3 input {
  width: 100%;
  padding: 0.5rem;
  background: inherit;
  border: none;
  font-size: 1rem;
}

.css-3 label {
  margin-bottom: 2px;
}

.css-3 label.fullWidth {
  width: 100%;
}

.css-3 label.hideLabel {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<body>
    <div>
      <div
        class="css-0"
      >
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
        <div
          class="css-3"
          lg="false"
        >
          <label
            class="hideLabel"
          >
            Search for gif
          </label>
          <input
            autocomplete="new-password"
            placeholder="Search"
            type="text"
            value="anger"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": .css-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 10px 16px;
}

.css-0>:not(style)+:not(style) {
  margin: 0;
  margin-left: 16px;
}

.css-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.css-1::-moz-focus-inner {
  border-style: none;
}

.css-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .css-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.css-1.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.css-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.css-3 {
  width: 315px;
  background: undefined;
  padding: 0.2rem 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom: 1px solid undefined;
  margin-bottom: 2rem;
}

.css-3 input {
  width: 100%;
  padding: 0.5rem;
  background: inherit;
  border: none;
  font-size: 1rem;
}

.css-3 label {
  margin-bottom: 2px;
}

.css-3 label.fullWidth {
  width: 100%;
}

.css-3 label.hideLabel {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
    <div
      class="css-0"
    >
      <button
        aria-label="close"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
      <div
        class="css-3"
        lg="false"
      >
        <label
          class="hideLabel"
        >
          Search for gif
        </label>
        <input
          autocomplete="new-password"
          placeholder="Search"
          type="text"
          value="anger"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<SearchModalHeader/> component : display <SearchModalHeader/> component  when Search result found  2`] = `
<DocumentFragment>
  .css-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 10px 16px;
}

.css-0>:not(style)+:not(style) {
  margin: 0;
  margin-left: 16px;
}

.css-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.css-1::-moz-focus-inner {
  border-style: none;
}

.css-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .css-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.css-1.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.css-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.css-3 {
  width: 315px;
  background: undefined;
  padding: 0.2rem 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom: 1px solid undefined;
  margin-bottom: 2rem;
}

.css-3 input {
  width: 100%;
  padding: 0.5rem;
  background: inherit;
  border: none;
  font-size: 1rem;
}

.css-3 label {
  margin-bottom: 2px;
}

.css-3 label.fullWidth {
  width: 100%;
}

.css-3 label.hideLabel {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div
    class="css-0"
  >
    <button
      aria-label="close"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
    <div
      class="css-3"
      lg="false"
    >
      <label
        class="hideLabel"
      >
        Search for gif
      </label>
      <input
        autocomplete="new-password"
        placeholder="Search"
        type="text"
        value="anger"
      />
    </div>
  </div>
</DocumentFragment>
`;
